### PR TITLE
Add the config changes to handle CHT communication with the mediator

### PR DESCRIPTION
**Pull request by njogz**
# Description

This adds the relevant config changes required for the CHT to handle communication with the mediator for the loss to followup workflow.

Closes medic/interoperability#13

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.
 
# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: cht-config/app_settings.json [hunk 1]
```diff
@@ -0,0 +1,584 @@
...
+          "type": "basic",
+          "username": "interop-client",
+          "password_key": "openhim1"
+        }
+      },
+      "mapping": {
+        "resourceType": {
+          "expr": "'Patient'"
+        },
+        "gender": "doc.sex",
```
Comment (samuelimoisili) [inline cht-config/app_settings.json:460]: The mediator is expecting `sex` not `gender`.

File: cht-config/app_settings.json [hunk 2]
```diff
@@ -0,0 +1,584 @@
...
+          "expr": "[ { \"use\":\"official\", \"family\": doc.name , \"given\": [ doc.short_name ] } ]",
+          "optional": false
+        },
+        "identifier.use": {
+          "expr": "'usual'"
+        },
+        "identifier.value": {
+          "expr": "'cht/'.concat(doc._id)"
+        },
+        "birthDate": "doc.date_of_birth",
```
Comment (samuelimoisili) [inline cht-config/app_settings.json:471]: We are expecting `date_of_birth` not `birthDate`. We had a discussion yesterday about data transformation taking place on the mediator and not cht-core.

File: cht-config/app_settings.json [hunk 3]
```diff
@@ -0,0 +1,584 @@
...
+          "expr": "[{ \"system\":\"phone\", \"value\": doc.phone, \"use\":\"mobile\" } ]",
+          "optional": true
+        },
+        "active": {
+          "expr": true,
+          "optional": false
+        }
+      }
+    },
+    "FHIR_practitioner": {
```
Comment (samuelimoisili) [inline cht-config/app_settings.json:482]: Do we need an issue to create the practitioner endpoint?

File: cht-config/app_settings.json [hunk 4]
```diff
@@ -0,0 +1,584 @@
...
+        "resourceType": {
+          "expr": "'ServiceRequest'"
+        },
+        "status": {
+          "expr": "'active'"
+        },
+        "intent": {
+          "expr": "'order'"
+        },
+        "subject": {
```
Comment (samuelimoisili) [inline cht-config/app_settings.json:543]: The mediator is expecting `patient_id` instead of subject.

**Review by samuelimoisili — CHANGES_REQUESTED** [review]

**Review by samuelimoisili — APPROVED** [review]

**Review by andrablaj — COMMENTED** [review]
@njogz I tried to upload the config locally and bumped into several errors. Please see my comments on the [related issue](https://github.com/medic/interoperability/issues/13#issuecomment-1440266236) and the steps I followed to mitigate the errors.

File: README.md [hunk 5]
```diff
@@ -24,6 +24,11 @@ A first version of the project can be found in the [chis-interoperability](https
 
 ![](./docs/images/bad-client-screen.png) If everything is successful you should see this ![](./docs/images/good-client-screen.png)
 
+## Loading CHT configuration
+1. Go into the `cht-config` directory by running `cd cht-config`.
+1. Run `npm install` to install the dependencies.
+1. Ensure you have [cht-conf](https://www.npmjs.com/package/cht-conf) installed and Run `cht --local` to compile and upload that configurationto upload the app settings to your local CHT-core instance.
```
Comment (andrablaj) [inline README.md:30]: I suggest adding a step for specifying that an instance of CHT should run locally and [the link](https://docs.communityhealthtoolkit.org/apps/tutorials/local-setup/) for the local setup instructions.

Comment (andrablaj) [inline README.md:30]: 1. Ensure you have [cht-conf](https://www.npmjs.com/package/cht-conf) installed and Run `cht --local` to compile and upload that configurationto upload the app settings to your local CHT-core instance.
```suggestion
1. Ensure you have [cht-conf](https://www.npmjs.com/package/cht-conf) installed and run `cht --local` to compile and upload the app settings configuration to your local CHT instance.
```

**Review by andrablaj — CHANGES_REQUESTED** [review]
Thanks for specifying which files we should pay attention to. 

Sharing some comments about the config upload part and the local setup for now.

File: README.md [hunk 6]
```diff
@@ -24,6 +24,11 @@ A first version of the project can be found in the [chis-interoperability](https
...
+
```
Comment (lorerod) [inline README.md:31]: I suggest adding the steps for the reader to try the CHT communication with the mediator.

File: README.md [hunk 7]
```diff
@@ -24,6 +24,11 @@ A first version of the project can be found in the [chis-interoperability](https
 
 ![](./docs/images/bad-client-screen.png) If everything is successful you should see this ![](./docs/images/good-client-screen.png)
 
+## Loading CHT configuration
+1. Go into the `cht-config` directory by running `cd cht-config`.
+1. Run `npm install` to install the dependencies.
```
Comment (lorerod) [inline README.md:29]: I wasn't able to make this work as per instructions. 
The steps I followed to be able to load the config where:

1. Created a directory outside the interoperability project
2. Copied the content of /interoperability/cht-config to the directory created in 1.
3. Executed cht initialise-project-layout
4. Executed cht --url=[https://user:pass@****.my.local-ip.co](https://medic:****@%2A%2A%2A%2A.my.local-ip.co/) --accept-self-signed-certs

**Review by lorerod — CHANGES_REQUESTED** [review]

File: cht-config/app_settings.json [hunk 8]
```diff
@@ -0,0 +1,590 @@
...
+        "active": {
+          "expr": true,
+          "optional": false
+        }
+      }
+    },
+    "FHIR_practitioner": {
+      "relevant_to": "doc.type === 'person' && (doc.role == 'chw' || doc.role == 'nurse' )",
+      "destination": {
+        "base_url": "http://localhost:5001",
```
Comment (samuelimoisili) [inline cht-config/app_settings.json:485]: We are updating the fhir database without passing through the mediator.

File: cht-config/app_settings.json [hunk 9]
```diff
@@ -0,0 +1,590 @@
...
+          "expr": true,
+          "optional": false
+        }
+      }
+    },
+    "FHIR_encounter": {
+      "relevant_to": "doc.type === 'data_record' && doc.form === 'interop_follow_up'",
+      "destination": {
+        "base_url": "http://localhost:5001",
+        "path": "/fhir/Encounter",
```
Comment (samuelimoisili) [inline cht-config/app_settings.json:523]: We are updating FHIR directly without passing through the mediator. Do we want to allow CHT communicate directly FHIR database?

**Comment by andrablaj** [comment]
Thank you for the updates @njogz! I tested the changes and I confirm that the config was uploaded successfully, without any extra action or file change from my side.

Please address my `README` suggestions above before I can review the PR.

**Review by andrablaj — APPROVED** [review]
Managed to upload the config by following the steps  in the `README`.

A simple way to verify if the config is loaded correctly is to access a URL like `https://*****.my.local-ip.co/#/contacts/someUUID/report/interop_follow_up `and to retrieve correctly the follow up form.

**Pull request merged**